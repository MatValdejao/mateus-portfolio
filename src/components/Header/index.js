import React from "react";

function Header() {
    return (
        <header >
            <h1>
                <a data-testid="header" href="/ ">Mateus Valdejao</a>
            </h1>
        </header>
    )
}

export default Header