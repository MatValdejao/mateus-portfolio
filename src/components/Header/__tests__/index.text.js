import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "..";


afterEach(cleanup)

describe("Header component", () => {
    it("renders", () => {
        render(<Header/>)
    })

    it("matches snapshot", () => {
        const { asFragment } = render(<Header />)
        expect(asFragment()).toMatchSnapshot()
    })
})


it("links are visible", () => {
    render(<Header />)

    expect(screen.getByTestId("header")).toHaveTextContent("Mateus Valdejao")
})