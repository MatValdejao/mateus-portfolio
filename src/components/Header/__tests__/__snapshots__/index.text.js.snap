// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1>
      <a
        data-testid="header"
        href="/ "
      >
        Mateus Valdejao
      </a>
    </h1>
  </header>
</DocumentFragment>
`;
